dialogflow/create_document_test.py

```diff
@@ -15,50 +15,50 @@
 from __future__ import absolute_import
 
 import os
+from unittest import mock
 import uuid
 
 from google.cloud import dialogflow_v2beta1 as dialogflow
 import pytest
 
 import document_management
+import test_utils
 
 PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
-KNOWLEDGE_BASE_NAME = "knowledge_{}".format(uuid.uuid4())
-DOCUMENT_DISPLAY_NAME = "test_document_{}".format(uuid.uuid4())
+KNOWLEDGE_BASE_NAME = f"knowledge_{uuid.uuid4()}"
+KNOWLEDGE_BASE_ID = uuid.uuid4().hex[0:27]
+DOCUMENT_DISPLAY_NAME = f"test_document_{uuid.uuid4()}"
+MIME_TYPE = "text/html"
+KNOWLEDGE_TYPE = "FAQ"
+CONTENT_URI = "https://cloud.google.com/storage/docs/faq"
 
 
 @pytest.fixture(scope="function")
-def knowledge_base_id():
-    # Create a knowledge base to use in document management
-    client = dialogflow.KnowledgeBasesClient()
-    project_path = client.common_project_path(PROJECT_ID)
-    knowledge_base = dialogflow.KnowledgeBase(display_name=KNOWLEDGE_BASE_NAME)
-    response = client.create_knowledge_base(
-        parent=project_path, knowledge_base=knowledge_base
-    )
-    knowledge_base_id = response.name.split("/knowledgeBases/")[1].split("\n")[0]
-
-    yield knowledge_base_id
-
-    # Delete the created knowledge base
-    knowledge_base_path = client.knowledge_base_path(
-        PROJECT_ID, knowledge_base_id
-    )
-    request = dialogflow.DeleteKnowledgeBaseRequest(
-        name=knowledge_base_path, force=True
+def mock_create_document_operation():
+    return test_utils.create_mock_create_document_operation(
+        DOCUMENT_DISPLAY_NAME,
+        KNOWLEDGE_BASE_ID,
+        MIME_TYPE,
+        [getattr(dialogflow.Document.KnowledgeType, KNOWLEDGE_TYPE)],
+        CONTENT_URI,
     )
-    client.delete_knowledge_base(request=request)
 
 
-@pytest.mark.flaky(max_runs=5, min_passes=1)
-def test_create_document(capsys, knowledge_base_id):
-    document_management.create_document(
-        PROJECT_ID,
-        knowledge_base_id,
-        DOCUMENT_DISPLAY_NAME,
-        "text/html",
-        "FAQ",
-        "https://cloud.google.com/storage/docs/faq",
-    )
-    out, _ = capsys.readouterr()
-    assert DOCUMENT_DISPLAY_NAME in out
+def test_create_document(
+    capsys: pytest.CaptureFixture[str],
+    mock_create_document_operation: mock.MagicMock,
+):
+    with mock.patch(
+        "google.cloud.dialogflow_v2beta1.DocumentsClient.create_document",
+        mock_create_document_operation,
+    ):
+        document_management.create_document(
+            PROJECT_ID,
```

Comment:
Since we are mocking out the create_document service, we can replace PROJECT_ID with a MOCK_PROJECT_ID, that is not a real project.